### PR TITLE
Add some message types from subsystem definitions

### PR DESCRIPTION
**Pull request by rphmeier**

### CODE REVIEW

File: node/primitives/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+/// A statement, where the candidate receipt is included in the `Seconded` variant.
+#[derive(Debug, Clone, PartialEq, Encode, Decode)]
+pub enum Statement {
+	/// A statement that a validator seconds a candidate.
+	#[codec(index = "1")]
+	Seconded(AbridgedCandidateReceipt),
+	/// A statement that a validator has deemed a candidate valid.
+	#[codec(index = "2")]
+	Valid(Hash),
+	/// A statement that a validator has deeped a candidate invalid.
```
Comment (coriolinus) [inline node/primitives/src/lib.rs:40]:
```suggestion
	/// A statement that a validator has deemed a candidate invalid.
```

File: node/primitives/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		signing_context: &SigningContext,
+	) -> Result<(), ()> {
+		let validator = validators.get(self.sender as usize).ok_or(())?;
+		let payload = self.statement.signing_payload(signing_context);
+
+		if self.signature.verify(&payload[..], validator) {
+			Ok(())
+		} else {
+			Err(())
+		}
```
Comment (coriolinus) [inline node/primitives/src/lib.rs:90]:
```suggestion
		self.signature.verify(&payload[..], validator).as_result((), ())
```

Only half-serious, but [`as_result`](https://docs.rs/boolinator/2.4.0/boolinator/trait.Boolinator.html#tymethod.as_result)

Comment (rphmeier) [reply node/primitives/src/lib.rs:90]: Aw, I was hoping this was `std`. Bringing in a crate & a trait for this sounds...heavy.

File: node/messages/src/lib.rs [hunk 3]
```diff
@@ -35,23 +44,103 @@ pub enum OverseerSignal {
...
+	/// and the PoV.
+	Validate(
+		Hash,
+		AbridgedCandidateReceipt,
+		PoVBlock,
+		oneshot::Sender<Result<(), ValidationFailed>>,
+	),
+}
+
+// TODO [now]: network bridge
```
Comment (rphmeier) [inline node/messages/src/lib.rs:92]: still needs resolving

File: node/messages/src/lib.rs [hunk 4]
```diff
@@ -35,23 +45,155 @@ pub enum OverseerSignal {
...
+		AbridgedCandidateReceipt,
+		PoVBlock,
+		oneshot::Sender<Result<(), ValidationFailed>>,
+	),
+}
+
+/// Chain heads.
+///
+/// Up to `N` (5?) chain heads.
+pub struct View(Vec<Hash>);
```
Comment (rphmeier) [inline node/messages/src/lib.rs:96]:
```suggestion
pub struct View(pub Vec<Hash>);
```

Comment (rphmeier) [reply node/messages/src/lib.rs:96]: maybe rename to `NetworkView` also

File: node/messages/src/lib.rs [hunk 5]
```diff
@@ -35,23 +45,155 @@ pub enum OverseerSignal {
...
+/// Availability Distribution Message.
+pub enum AvailabilityDistributionMessage {
+	/// Distribute an availability chunk to other validators.
+	DistributeChunk(Hash, ErasureChunk),
+
+	/// Fetch an erasure chunk from networking by candidate hash and chunk index.
+	FetchChunk(Hash, u32),
+
+	/// Event from the network bridge.
+	NetworkBridgeEvent(NetworkBridgeEvent),
```
Comment (rphmeier) [inline node/messages/src/lib.rs:137]: The guide calls this `NetworkBridgeUpdate`

File: node/messages/src/lib.rs [hunk 6]
```diff
@@ -35,23 +45,155 @@ pub enum OverseerSignal {
...
+}
+
+/// Bitfield distribution message.
+pub enum BitfieldDistributionMessage {
+	/// Distribute a bitfield via gossip to other validators.
+	DistributeBitfield(Hash, SignedAvailabilityBitfield),
+
+	/// Event from the network bridge.
+	NetworkBridgeEvent(NetworkBridgeEvent),
```
Comment (rphmeier) [inline node/messages/src/lib.rs:146]: The guide calls this `NetworkBridgeUpdate`

**Comment by rphmeier** [comment]
Needs to be merged now that #1270 is in, and then is missing a couple message types. Otherwise LGTM

**Pull request merged**